Source Code/Ticket Management System/supportgiver_menu.c

```diff
@@ -25,11 +25,38 @@ bool filterByStatus(void* ticket, void* status)
 {
 	return wcscmp(ticket_getStatus(ticket), status) != 0;
 }
+bool filterByTier(void* ticket1, void* ticket2)
+{
+	return wcscmp(ticket_getTier(ticket1), ticket_getTier(ticket2)) != 0;
+}
+bool filterByType(void* ticket1, void* ticket2)
+{
+	return wcscmp(ticket_getType(ticket1), ticket_getType(ticket2)) != 0;
+
+}
+bool sortBySeverity(void* ticket1, void* ticket2)
+{
+	return wcscmp(ticket_getSeverity(ticket1), ticket_getSeverity(ticket2)) > 0;
 
+}
 bool sortByTitle(void* ticket1, void* ticket2)
 {
 	return wcscmp(ticket_getTitle(ticket1), ticket_getTitle(ticket2)) > 0;
 }
+bool filterByStakeholders(void* ticket1, void*ticket2)
+{
+	return wcscmp(ticket_getstakeholders(ticket1), ticket_getstakeholders(ticket2))!= 0;
+}
+bool searchByCustomerEmail(void* ticket1, void*ticket2)
+{
+	return wcscmp(ticket_getCustomerEmail(ticket1), ticket_getCustomerEmail(ticket2)) != 0;
+
+}
+bool searchBytags(void* ticket1, void*ticket2)
+{
+	return wcscmp(ticket_getTags(ticket1), ticket_getTags(ticket2)) != 0;
```

Comment:
Why `searchBytags` gets 2 tickets?
Why not ticket and tags?

Reply:
Still here.